llama.cpp

```diff
@@ -9425,6 +9464,9 @@ struct llama_context * llama_new_context_with_model(
 
 void llama_free(struct llama_context * ctx) {
     delete ctx;
+#ifdef GGML_USE_KOMPUTE
+    ggml_vk_free_device();
+#endif
```

Comment:
This should go in `llama_backend_free()`

Reply:
It should be done automatically after all the backend and buffer objects are freed. Metal buffers do something similar.

Reply:
This is already done in ggml_backend_kompute_free, so I can remove this one. Is there any reason I should prefer the way Metal does it?

Reply:
The way Metal does it allows applications to free all the resources when the backend is no longer in use. It is not a big issue regardless.

Reply:
However, if this will cause the Kompute backend to break if there are multiple instances of the backend and one is freed, that would be an issue.

Reply:
If we don't know which GPU we are supposed to be using (because we will be "do[ing] the initialization automatically on the default device only if it was not already initialized by a call to ggml_vk_init_device"), it's not so straightforward to eagerly free the GPU and lazily re-initialize it later.